fix(ci): Deployment workflow fails at static IP assignment due to read command exit code

## Describe the issue or request

The deployment workflow fails during the "Assign static IPs to instances" step when executing the `read -rd ''` command. This command always returns exit code 1 when it encounters EOF, even though it successfully populates the array. Because GitHub Actions runs bash with `set -e`, this non-zero exit code causes the entire workflow step to fail.

## Expected Behavior

The workflow should continue after successfully populating the `INSTANCE_ROWS` array, proceeding to assign static IPs to the instances.

## Current Behavior

The workflow fails immediately after the `read -rd ''` command completes:

```
Waiting for 3 instances to be created...
All 3 instances exist
Error: Process completed with exit code 1.
```

The failure occurs at line 447 in `.github/workflows/zfnd-deploy-nodes-gcp.yml`:

```bash
IFS=$'\n' read -rd '' -a INSTANCE_ROWS <<<"$INSTANCE_DATA"
```

## Possible Solution

Add `|| true` to the read command to allow the script to continue since the array is correctly populated:

```bash
IFS=$'\n' read -rd '' -a INSTANCE_ROWS <<<"$INSTANCE_DATA" || true
```

This is a known behavior of the `read` builtin when using the `-d ''` delimiter.

## Additional Information/Context

- Workflow run: https://github.com/ZcashFoundation/zebra/actions/runs/19448245778/job/55716322494
- Failed at: Deploy Mainnet nodes step
- Error occurred: 2025-11-18T14:59:54Z
- The array is successfully populated before the exit code causes failure

## Is this happening on PRs?

No, this occurs during deployments triggered by main branch pushes or releases.

## Is this happening on the main branch?

Yes, this affects deployments from the main branch.

--- a/.github/workflows/zfnd-deploy-nodes-gcp.yml
+++ b/.github/workflows/zfnd-deploy-nodes-gcp.yml
@@ -444,7 +444,7 @@ jobs:
             --region "${{ vars.GCP_REGION }}" \
             --format="table[no-heading](instance.basename(),instance.scope().segment(0))" | sort)
 
-          IFS=$'\n' read -rd '' -a INSTANCE_ROWS <<<"$INSTANCE_DATA"
+          IFS=$'\n' read -rd '' -a INSTANCE_ROWS <<<"$INSTANCE_DATA" || true
 
           # Loop to assign all IPs
           for i in "${!INSTANCE_ROWS[@]}"; do

